### DISCUSSION TITLE
I can't figure out how to add serverside translations

### DISCUSSION THREAD

**Discussion by Moise-VK** [nuxt/nuxt#25606] [Questions] [answered]
I want to use nuxt for serverside rendering. Now I also want to use i18n for translations. I saw an experimental feature for serverside translations but at the end I need to add useTranslation on eventhandler and it's there I don't really know what to do. Could someone please help me?

The documentation I'm talking about can be found here: https://i18n.nuxtjs.org/guide/server-side-translations#usetranslation-on-eventhandler

**Comment by undermoonn** [comment, marked as answer]
You can check more detail of use case in https://vue-i18n.intlify.dev/guide/

`nuxt/i18n` provides i18n setup in `nuxt.config.ts`, after setting that you can use `useI18n` in script or `$t` in vue template

```vue
<script setup lang="ts">
const { t } = useI18n()
const title = computed(() => t('locale_title_key'))
</script>
```

or
```vue
<template>
  <h1>{{ $t('locale_title_key') }}</h1>
</template>
```

**Reply by Moise-VK** [reply]
Ah yes I checked the pagesource and I indeed saw everything was in there. so this -> https://i18n.nuxtjs.org/guide/server-side-translations#usetranslation-on-eventhandler <- wasn't necessary. Thank you!

**Reply by Zyles** [reply]
> You can check more detail of use case in https://vue-i18n.intlify.dev/guide/
> 
> `nuxt/i18n` provides i18n setup in `nuxt.config.ts`, after setting that you can use `useI18n` in script or `$t` in vue template
> 
> ```vue
> <script setup lang="ts">
> const { t } = useI18n()
> const title = computed(() => t('locale_title_key'))
> </script>
> ```
> 
> or
> 
> ```vue
> <template>
>   <h1>{{ $t('locale_title_key') }}</h1>
> </template>
> ```

Link leads to 404.

Is there a solution to run "generate" with language set by ENV variable?

Nothing I have tried works.
